src/GeneratedHydrator/CodeGenerator/Visitor/HydratorMethodsVisitor.php

```diff
@@ -133,7 +133,7 @@ private function replaceConstructor(ClassMethod $method)
             // Hydrate closures
             $bodyParts[] = "\$this->hydrateCallbacks[] = \\Closure::bind(function (\$object, \$values) {";
             foreach ($propertyNames as $propertyName) {
-                $bodyParts[] = "    if (isset(\$values['" . $propertyName . "'])) {";
+                $bodyParts[] = "    if (isset(\$values['$propertyName']) || \$object->$propertyName !== null && array_key_exists('$propertyName', \$values)) {";
```

Comment:
It's very visually different, especially when reading through code and not knowing whether the symbols will be interpolated or not. That's extremely relevant in codegen, so I got rid of all interpolation in ProxyManager too, for example, as it otherwise makes things much more confusing.